conn.go

```diff
@@ -166,9 +166,15 @@ func (s *Swarm) addConn(netConn net.Conn, isServer bool) (*Conn, error) {
 	// unlocked **before** the Handler is called (which needs to be
 	// sequential). This was the simplest thing :)
 	setupConn := func() (*Conn, error) {
+
+		// create a new spdystream connection
+		ssConn, err := s.transport.NewConn(netConn, isServer)
+		if err != nil {
+			return nil, err
+		}
+
```

Comment:
if doing things that way, we should put a placeholder connection in the map until we finish the handshake, that way we dont get any weird race conditions.